bbc/digital-paper-edit-firebase
# #176 Fixes corrupted media file downloads

**Is your Pull Request request related to another [issue](https://github.com/bbc/react-transcript-editor/issues) in this repository ?**      

#176 

**Describe what the PR does**    

- [x] Passes file type to the downloadjs function, which fixed media files downloading as uris
- [x] Opens a stream to ensure the whole media file is loaded before being saved locally (fixes the corrupted files warnings)
- [x] Lightly refactors the MediaModal component to give feedback on download progress

<img width="1269" alt="Screenshot 2021-04-08 at 18 39 37" src="https://user-images.githubusercontent.com/22618458/114072171-dc65a800-9899-11eb-8cd7-b774a7db8d3b.png">

**State whether the PR is ready for review or whether it needs extra work**    

Ready for review 

**Additional context**    

To test:

1. Navigate to a PaperEdit
2. Click 'Export'
3. Try to export all the files and then open them and confirm you aren't getting any corrupted file warnings, etc

I've validated this on my end but please check it's fixed with any corrupted files you've noticed!
